webserver/cors/src/main/java/io/helidon/webserver/cors/CrossOriginConfig.java

```diff
@@ -0,0 +1,308 @@
+        this.maxAge = builder.maxAge;
+    }
+
+    /**
+     * @return a new builder for basic cross origin config
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Creates a new {@code CrossOriginConfig} instance using the provided config node.
+     *
+     * @param config node containing cross-origin information
+     * @return new {@code Basic} instance based on the configuration
+     */
+    public static CrossOriginConfig from(Config config) {
+        return Builder.from(config).build();
+    }
+
+    /**
+     * @return the configured path prefix; defaults to a "match-everything" pattern
+     */
+    public String pathPrefix() {
+        return pathPrefix;
+    }
+
+    /**
+     * @return whether this cross-origin config is enabled
+     */
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    /**
+     * @return the allowed origins
+     */
+    public String[] allowOrigins() {
+        return copyOf(allowOrigins);
+    }
+
+    /**
+     * @return the allowed headers
+     */
+    public String[] allowHeaders() {
+        return copyOf(allowHeaders);
+    }
+
+    /**
+     * @return headers OK to expose in responses
+     */
+    public String[] exposeHeaders() {
+        return copyOf(exposeHeaders);
+    }
+
+    /**
+     * @return allowed methods
+     */
+    public String[] allowMethods() {
+        return copyOf(allowMethods);
+    }
+
+    /**
+     * @return allowed credentials
+     */
+    public boolean allowCredentials() {
+        return allowCredentials;
+    }
+
+    /**
+     * @return maximum age
+     */
+    public long maxAge() {
+        return maxAge;
+    }
+
+    private static String[] copyOf(String[] strings) {
+        return strings != null ? Arrays.copyOf(strings, strings.length) : new String[0];
+    }
+
+    /**
+     * Builder for {@link CrossOriginConfig}.
+     */
+    public static class Builder implements Setter<Builder>, io.helidon.common.Builder<CrossOriginConfig>,
+            Function<Config, Builder> {
+
+        static final String[] ALLOW_ALL = {"*"};
+
+        private String pathPrefix = PATHLESS_KEY; // not typically used except when inside a MappedCrossOriginConfig
+        private boolean enabled = true;
+        private String[] origins = ALLOW_ALL;
+        private String[] allowHeaders = ALLOW_ALL;
+        private String[] exposeHeaders;
+        private String[] allowMethods = ALLOW_ALL;
+        private boolean allowCredentials;
+        private long maxAge = DEFAULT_AGE;
+
+        private Builder() {
+        }
+
+        /**
+         * Creates a new builder based on the values in an existing {@code CrossOriginConfig} object.
+         *
+         * @param original the existing cross-origin config object
+         * @return new Builder initialized from the existing object's settings
+         */
+        public static Builder from(CrossOriginConfig original) {
```

Comment:
Please do not add factory methods to builders themself.
Either do `CrossOriginConfig.builder(CrossOriginConfig config)` or add an instance method `config(CrossOriginConfig)`

Reply:
See below